src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -164,6 +173,21 @@ public String getDedupeBy() {
     return Strings.isNullOrEmpty(dedupeBy) ? null : dedupeBy;
   }
 
+  @Nullable
+  public String getPartitionFilter() {
+    if (partitionFilter != null) {
+      partitionFilter = partitionFilter.trim();
+      if (partitionFilter.isEmpty()) {
+        return null;
+      }
+      // remove the WHERE from the filter
+      if (partitionFilter.toUpperCase().startsWith(WHERE)) {
+        partitionFilter = partitionFilter.replace(WHERE, "");
+      }
+    }
+    return partitionFilter;
```

Comment:
better have it reverse
```
if (Strings.isNullOrEmpty(partitionFilter)) {
 return partitionFilter;
}
...

Reply:
@CuriousVini will refactor the above to:

```
if (Strings.isNullOrEmpty(partitionFilter)) {
  return null;
}
partitionFilter = partitionFilter.trim();
// remove the WHERE keyword from the filter if the user adds it at the begging of the expression
if (partitionFilter.toUpperCase().startsWith(WHERE)) {
   partitionFilter = partitionFilter.substring(WHERE.length());
}
return  partitionFilter;
```